src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -221,6 +221,10 @@ void ApplyEnrichment(LoggerConfiguration loggerConfiguration, IReadOnlyDictionar
             IConfigurationArgumentValue GetArgumentValue(IConfigurationSection argumentSection)
             {
                 IConfigurationArgumentValue argumentValue;
+
+                if (argumentSection.Value != null && argumentSection.GetChildren().Any())
```

Comment:
now you've explained it, this precondition check absolutely belongs here on its own and not nested